src/components/withdrawals/CreateForm.tsx

```diff
@@ -0,0 +1,219 @@
+import React, { useState } from 'react'
+import { useMutation } from 'react-query'
+import { useRouter } from 'next/router'
+import { useTranslation } from 'next-i18next'
+import Link from 'next/link'
+import { AxiosError, AxiosResponse } from 'axios'
+import {
+  Box,
+  Button,
+  FormControl,
+  FormLabel,
+  Grid,
+  Input,
+  MenuItem,
+  Select,
+  Typography,
+} from '@mui/material'
+
+import { WithdrawalInput, WithdrawalResponse } from 'gql/withdrawals'
+import {
+  useBankAccountsForWithdrawal,
+  useCampaignsForWithdrawal,
+  usePersonListForWithdrawal,
+  useVaultsForWithdrawal,
+} from 'common/hooks/withdrawals'
+import { routes } from 'common/routes'
+import { ApiErrors } from 'service/apiErrors'
+import { useCreateWithdrawal } from 'service/withdrawal'
+import { AlertStore } from 'stores/AlertStore'
+import SubmitButton from 'components/common/form/SubmitButton'
+import { WithdrawalStatus } from './WithdrawalTypes'
+
+export default function CreateForm() {
+  const router = useRouter()
+  const bankAccounts = useBankAccountsForWithdrawal()
+  const vaults = useVaultsForWithdrawal()
+  const campaigns = useCampaignsForWithdrawal()
+  const personList = usePersonListForWithdrawal()
+  const currencies = ['BGN', 'USD', 'EUR']
+  const [currency, setCurrency] = useState('')
+  const [amount, setAmount] = useState(Number)
+  const [reason, setReason] = useState('')
+  const [approvedById, setApprovedById] = useState('')
+  const [bankAccountId, setBankAccountId] = useState('')
+  const [vaultId, setVaultId] = useState('')
+  const [campaignId, setCampaignId] = useState('')
+
+  const { t } = useTranslation()
+
+  const mutationFn = useCreateWithdrawal()
+
+  const mutation = useMutation<
+    AxiosResponse<WithdrawalResponse>,
+    AxiosError<ApiErrors>,
+    WithdrawalInput
+  >({
+    mutationFn,
+    onError: () => AlertStore.show(t('withdrawals:alerts:error'), 'error'),
+    onSuccess: () => {
+      AlertStore.show(t('withdrawals:alerts:create'), 'success')
+      router.push(routes.admin.withdrawals.index)
+    },
+  })
+
+  function handleSubmit() {
+    const data: WithdrawalInput = {
+      status: WithdrawalStatus.initial,
+      currency: currency,
+      amount: amount,
+      reason: reason,
+      sourceVaultId: vaultId,
+      sourceCampaignId: campaignId,
+      bankAccountId: bankAccountId,
+      documentId: '6061b425-34e6-45bf-b563-a85b933c9339',
```

Comment:
If `documentId` is mandatory maybe we should make it optional

If it's not mandatory then we don't need to pass it at the stage of withdrawal creation as it comes later on in real life